actortoken/actor_token.go

```diff
@@ -0,0 +1,42 @@
+// Package actortoken provides the Actor Tokens API.
+package actortoken
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+)
+
+const path = "/actor_tokens"
+
+type CreateParams struct {
+	clerk.APIParams
+	UserID                      *string         `json:"user_id,omitempty"`
+	Actor                       json.RawMessage `json:"actor,omitempty"`
+	ExpiresInSeconds            *int64          `json:"expires_in_seconds,omitempty"`
+	SessionMaxDurationInSeconds *int64          `json:"session_max_duration_in_seconds,omitempty"`
+}
+
+// Create creates a new actor token.
+func Create(ctx context.Context, params *CreateParams) (*clerk.ActorToken, error) {
+	req := clerk.NewAPIRequest(http.MethodPost, path)
+	req.SetParams(params)
+	token := &clerk.ActorToken{}
+	err := clerk.GetBackend().Call(ctx, req, token)
+	return token, err
+}
+
+// Revoke revokes a pending actor token.
+func Revoke(ctx context.Context, id string) (*clerk.ActorToken, error) {
+	path, err := url.JoinPath(path, id, "revoke")
+	if err != nil {
+		return nil, err
+	}
+	req := clerk.NewAPIRequest(http.MethodPost, path)
+	token := &clerk.ActorToken{}
+	err = clerk.GetBackend().Call(ctx, req, token)
+	return token, err
```

Comment:
❓ Do you want to return `nil` or error or a pointer to empty struct? Because as it is now, it seems more suitable to return the empty value instead of a pointer.

Reply:
You mean because we return either nil or the pointer and we should be consistent and always return one or the other in case of error?

Reply:
Usually with pointers as return values we see in the code something like:

```go
if err != nil {
  return nil, err
}
```

Here we return a pointer to empty struct which is not `nil`. Technically, non-nil error values should always take precedence at the call site. I am personally in favor for returning values not pointers:

```go
return clerk.ActorToken{}, err
```